activation/wire/wire_v1.go

```diff
@@ -173,10 +173,10 @@ func ActivationTxFromBytes(data []byte) (*types.ActivationTx, error) {
 		return nil, fmt.Errorf("decoding ATX: %w", err)
 	}
 
-	return ActivationTxFromWireV1(&wireAtx), nil
+	return ActivationTxFromWireV1(&wireAtx, data...), nil
 }
 
-func ActivationTxFromWireV1(atx *ActivationTxV1) *types.ActivationTx {
+func ActivationTxFromWireV1(atx *ActivationTxV1, blob ...byte) *types.ActivationTx {
```

Comment:
Ideally this function should only be called in the handler and no where else (just before storing the ATX in the DB) - same for `ActivationTxFromBytes` above.

The `ActivationTxToWireV1` should in my opinion not be needed at all after we've finished the refactoring there should never be need to convert an domain ATX into a wire type: gossip and sync just serve the raw bytes from the DB and other parts of the code base should either only use the wire type (builder, handler) or only the domain type (basically everything else).